diff --git a/src-c/native/bitfn.h b/src-c/native/bitfn.h
index 8ce6216..b025b04 100644
--- a/src-c/native/bitfn.h
+++ b/src-c/native/bitfn.h
@@ -259,6 +259,9 @@ static __INLINE void store64( void *dst, uint64_t w )
 #elif defined( __QNXNTO__ ) && defined( __BIGENDIAN__ )
   # define BIG_ENDIAN 1234
   # define BYTE_ORDER    BIG_ENDIAN
+#elif defined(_MSC_VER)
+  # define LITTLE_ENDIAN	1234
+  # define BYTE_ORDER	    LITTLE_ENDIAN
 #else
   # include <endian.h>
 #endif
diff --git a/src-c/native/digestif.h b/src-c/native/digestif.h
index 65c436f..c7e4570 100644
--- a/src-c/native/digestif.h
+++ b/src-c/native/digestif.h
@@ -32,7 +32,14 @@
 #endif
 
 #ifndef __unused
-#define __unused(x) x __attribute__((unused))
+# if defined(_MSC_VER) && _MSC_VER >= 1500
+#  define __unused(x) __pragma( warning (push) ) \
+    __pragma( warning (disable:4189 ) ) \
+    x \
+    __pragma( warning (pop))
+# else
+#  define __unused(x) x __attribute__((unused))
+# endif
 #endif
 #define __unit() value __unused(unit)
 
