diff --git a/configure.make b/configure.make
index ab585cd..3934989 100644
--- a/configure.make
+++ b/configure.make
@@ -92,10 +92,10 @@ Makefile.config:
 src/ocamlbuild_config.ml:
 	(echo "(* This file was generated from ../configure.make *)"; \
 	echo ;\
-	echo 'let bindir = "$(OCAMLBUILD_BINDIR)"'; \
-	echo 'let libdir = "$(OCAMLBUILD_LIBDIR)"'; \
-	echo 'let ocaml_libdir = "$(abspath $(OCAML_LIBDIR))"'; \
-	echo 'let libdir_abs = "$(abspath $(OCAMLBUILD_LIBDIR))"'; \
+	echo 'let bindir = {|$(OCAMLBUILD_BINDIR)|}'; \
+	echo 'let libdir = {|$(OCAMLBUILD_LIBDIR)|}'; \
+	echo 'let ocaml_libdir = {|$(abspath $(OCAML_LIBDIR))|}'; \
+	echo 'let libdir_abs = {|$(abspath $(OCAMLBUILD_LIBDIR))|}'; \
 	echo 'let ocaml_native = $(OCAML_NATIVE)'; \
 	echo 'let ocaml_native_tools = $(OCAML_NATIVE_TOOLS)'; \
 	echo 'let supports_shared_libraries = $(SUPPORTS_SHARED_LIBRARIES)';\
